xapian-letor/tests/api_letor.cc

```diff
@@ -182,6 +182,49 @@ DEFINE_TESTCASE(svm_ranker, generated)
     return true;
 }
 
+DEFINE_TESTCASE(listmle_ranker, generated)
+{
+    Xapian::ListMLERanker ranker;
+    TEST_EXCEPTION(Xapian::FileNotFoundError, ranker.train_model(""));
+    string db_path = get_database_path("apitest_listmle_ranker",
+				       db_index_two_documents);
+    Xapian::Enquire enquire((Xapian::Database(db_path)));
+    enquire.set_query(Xapian::Query("lions"));
+    Xapian::MSet mymset = enquire.get_mset(0, 10);
+    string data_directory = test_driver::get_srcdir() + "/testdata/";
+    string query = data_directory + "query.txt";
+    string qrel = data_directory + "qrel.txt";
+    string training = test_driver::get_srcdir() + "/" +
+	    "listmle_ranker_training.txt";
```

Comment:
James mentioned on IRC, that you should combine the two strings - in C++ `x + "/" + "listmle_ranker_training.txt"` ends up being done as two separate string concatenations.

Which is a fair point if you're going to concatenate strings here, but actually, this is a generated file, so you should go in the build directory (i.e. the current directory) and not in `srcdir`.  When the build directory != `srcdir`, you should be able to run a build with a read-only `srcdir`.  This allows configuring the same source tree multiple ways, so you can test different build configurations in parallel.

It would be a good idea to read up on this a bit - see:

https://www.gnu.org/software/automake/manual/html_node/VPATH-Builds.html

This testcase should work like the one using `"listnet_ranker_training.txt"`, which is correctly generated in the build directory.